src/TenancyServiceProvider.php

```diff
@@ -69,6 +70,12 @@ public function register(): void
             return new Commands\Seed($app['db']);
         });
 
+        // Use custom mail manager that resolves the mailers specified in its $tenantMailers static property
+        // Instead of getting the cached mailers from the $mailers property
+        $this->app->extend(MailManager::class, function (MailManager $mailManager) {
+            return new TenancyMailManager($this->app);
+        });
```

Comment:
I wonder if this could be moved to the bootstrapper?

Reply:
I like that idea. It works with a small change – tenancy has to be initialized for the bootstrapper to be bootstrapped (so that MailManager resolves as TenancyMailManager), which makes sense to me.

Reply:
Yeah, that is correct.

I guess moving it to the bootstrapper comes with a _small_ issue — there may be an edge case where people interact with the mailer (or try to inject it somewhere, thus triggering its instantiation from the container) before the bootstrapper is executed. But if that ever comes up, we can just tell people to put this in a service provider.

Reply:
> if that ever comes up, we can just tell people to put this in a service provider

Agree

Reply:
For the record, I'm resolving this because I added a note about mentioning that in the PR's description